src/muya/lib/contentState/copyCutCtrl.js

```diff
@@ -182,7 +182,41 @@ const copyCutCtrl = ContentState => {
     return { html: htmlData, text: textData }
   }
 
-  ContentState.prototype.copyHandler = function (event, type) {
+  ContentState.prototype.copyHandler = function (event, type, copyInfo = null) {
+    if (event.target.tagName === 'BODY') {
+      const { selectedTableCells } = this
+      if (selectedTableCells) {
+        event.preventDefault()
+        const { row, column, cells } = selectedTableCells
+        const figureBlock = this.createBlock('figure', {
+          functionType: 'table'
+        })
+        const tableContents = []
+        let i
+        let j
```

Comment:
I think declare variable before it is been use is a good practice. if we put `let` in the `for loop`, it will declare the `i` every turn when the for loop run.

Reply:
Declaring the variable before is only a thing in `C` due to language restrictions but don't give better performance because all compilers/VMs compile a for loop to something below. Declaring variables outside the loop makes sense when you don't reallocate it during loops or used as buffer because it's only allocated once and cached in registers.

```
  set i := 0
:loop // label
  nop // code inside loop
  inc i
  cmp i, <end-var> // jump to loop if true
  nop
```

Reply:
Yeh but there already have declare `i` out of for loop in many spaces .🤣, also for unify the code style.

Reply:
Declaring the loop variable inside or outside have the same performance but use what you want. Different story is `i++` vs `++i` because the later is atomic on x64 CPUs whereas `i++` not.